charts/observability-pipelines-worker/README.md

```diff
@@ -20,15 +20,15 @@ To install the chart with the release name `<RELEASE_NAME>` run:
 ```bash
 helm install --name <RELEASE_NAME> \
   --set datadog.apiKey=<DD_API_KEY> \
-  --set datadog.configKey=<DD_OP_CONFIG_KEY> \
+  --set datadog.pipelineID=<DD_OP_PIPELINE_ID> \
   datadog/observability-pipelines-worker
 ```
 
-By default, this chart creates secrets for your Observability Pipelines API and configuration keys. However, you can use
-manually created Secrets by setting the `datadog.apiKeyExistingSecret` and/or `datadog.appKeyExistingSecret` values
+By default, this chart creates secrets for your Observability Pipelines API key and pipeline ID. However, you can use
```

Comment:
@spencergilbert I made some updates in [48707e1](https://github.com/DataDog/helm-charts/pull/952/commits/48707e1d02ddbb51ac958161105f9a6bd2ad0280) and [30d5b0a](https://github.com/DataDog/helm-charts/pull/952/commits/30d5b0ad0c8c283bee223fda61379c252c671cab). Let me know how that looks.

I'm curious if we absolutely need secrets management for the config key/pipeline ID? It's essentially just a UUID identifier for a particular pipeline, so I'm not sure it's considered sensitive data. This is mostly for my own understanding, I think we're fine to leave as is for now.

Reply:
If it's just a UUID, it may not need to be a Secret. Could anything funky happen if an attacker reused the same ID?

I think I was under the impression that it was suppose to be a confidential value when I wrote this, if not it could just as easily be a ConfigMap.

Reply:
> Could anything funky happen if an attacker reused the same ID?

Off the top of my head, I don't think anything malicious can be done with solely the pipeline ID. 

> I think I was under the impression that it was suppose to be a confidential value when I wrote this, if not it could just as easily be a ConfigMap.

Right, I'm wondering if we could treat it like the other values under `datadog`. Perhaps we could simplify this in a follow up? It doesn't seem particularly urgent.

Reply:
Ah, yeah if it's determined there's not an issue exposing it we could transition it to similar to the `datadog.site` or `datadog.url` options.

I'd want to doublecheck the assumption before making that happen, for sure.

Reply:
> Off the top of my head, I don't think anything malicious can be done with solely the pipeline ID.

Just a drive-by...

Pipeline ID isn't intended to be sensitive. It's really just a resource ID, embedded in URLs and made public in UI without regard to obfuscation. Doesn't need to be a secret.

Reply:
Definitely sounds like something to change then, apologies for my misunderstanding!

Reply:
@spencergilbert, for simplicity's sake, how do we feel about
- Treat `pipelineId` similar to `datadog.___` options
- Remove the `configKey` secret entirely

as part of this release?

Reply:
No objections for the first. For the second, that may be considered a breaking change if users had been using the `existingConfigKey` options.

It would complicate the code a here a bit, but we could do `datadog.pipelinesId` or `datadog.configKey` goes directly in the Pod's `env` block (both creating the _new_ env var name), and `datadog.configKeyExistingSecret` still using a referenced Secret to populate it. Marking the two `configKey` options as deprecated and dropping next release?

Reply:
> No objections for the first.

Ok, I'll make some minor updates then for `pipelineId`.

> It would complicate the code a here a bit, but we could do datadog.pipelinesId or datadog.configKey goes directly in the Pod's env block (both creating the new env var name), and datadog.configKeyExistingSecret still using a referenced Secret to populate it. Marking the two configKey options as deprecated and dropping next release?

Sure, this seems reasonable. Would you be able to make the updates here?

To make sure I'm on the right page, we're planning to set `DD_OP_PIPELINE_ID` in the environment in order of precedence from 1) `datadog.pipelineId` 2) `datadog.configKey` 3) `datadog.configKeyExistingSecret`? And we can get rid of some current plumbing that creates a secret from `datadog.configKey`?

Reply:
Yeah, I can make some time tonight to do that. Your understanding sounds right, though I'll double check that there's no issues with `datadog.configKey` no longer creating a Secret - between versions.

Reply:
Pushed some changes in https://github.com/DataDog/helm-charts/pull/952/commits/34e8658df19b73bed271faa39d5b2c7036313c65, I left the secret as-is since I don't think Helm will natively remove the orphaned resources.